docs/fundamentals/code-analysis/style-rules/language-rules.md

```diff
@@ -0,0 +1,1920 @@
+---
+title: Code style language rules
+ms.date: 09/25/2020
+ms.topic: reference
+author: gewarren
+ms.author: gewarren
+dev_langs:
+- CSharp
+- VB
+helpviewer_keywords:
+- language code style rules [EditorConfig]
+- language rules
+- EditorConfig language conventions
+---
+# Language rules
+
+Code style language rules affect how various constructs of .NET programming languages are used, for example, modifiers and parentheses. They fall into the below categories:
+
+- [.NET style rules](#net-style-rules): Rules that apply to both C# and Visual Basic. The EditorConfig option names for these rules start with `dotnet_style_` prefix.
+- [C# style rules](#c-style-rules): Rules that are specific to C# language only. The EditorConfig option names for these rules start with `csharp_style_` prefix.
+
+> [!TIP]
+>
+> - To see the code examples in your preferred programming language, choose it using the language picker at the top-right corner of the browser window.
+>
+>   ![Code language picker control](media/code-language-picker.png)
+>
+> - Use the **In this article** links to jump to different sections of the page.
+
+## Option format
+
+Options for language rules can be specified in an EditorConfig file with the following format:
+
+`option_name = value:severity`
+
+- **Value**: For each language rule, you specify a value that defines if or when to prefer the style. Many rules accept a value of `true` (prefer this style) or `false` (do not prefer this style). Other rules accept values such as `when_on_single_line` or `never`.
+- **Severity**: The second part of the rule specifies the [severity level](../configuration-options.md#severity-level) for the rule. Severity specification as part of the above option syntax is only respected inside development IDEs, such as Visual Studio. This setting is not understood by the C# or VB compilers, hence not respected during build. Instead, to enforce code style rules on build, you should set the severity by using the rule ID based severity configuration syntax for analyzers. The syntax takes the form `dotnet_diagnostic.<rule ID>.severity = <severity>`, for example, `dotnet_diagnostic.IDE0040.severity = silent`. For more details, see this [github issue](https://github.com/dotnet/roslyn/issues/44201).
+
+> [!TIP]
+>
+> Starting in Visual Studio 2019 version 16.3, you can configure code style rules from the [Quick Actions](/visualstudio/ide/quick-actions.md) light bulb menu after a style violation occurs. For more information, see [Automatically configure code styles in Visual Studio](/visualstudio/ide/editorconfig-language-conventions#automatically-configure-code-styles-in-visual-studio)
+
+## .NET style rules
```

Comment:
Content from here on is mostly identical to the original doc and is the core index/rule reference.